src/Vision/VisionClient.php

```diff
@@ -0,0 +1,126 @@
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Vision;
+
+use Google\Cloud\ClientTrait;
+use Google\Cloud\Storage\Object as StorageObject;
+use Google\Cloud\Vision\Connection\Rest;
+use InvalidArgumentException;
+
+/**
+ * Google Cloud Vision client. Allows you to send and receive
+ * messages between independent applications. Find more information at
+ * [Google Cloud Vision docs](https://cloud.google.com/vision/docs/).
+ */
+class VisionClient
+{
+    use ClientTrait;
+
+    const FULL_CONTROL_SCOPE = 'https://www.googleapis.com/auth/cloud-platform';
+
+    /**
+     * @var ConnectionInterface
+     */
+    protected $connection;
+
+    /**
+     * Create a Vision client.
+     *
+     * Example:
+     * ```
+     * use Google\Cloud\ServiceBuilder;
+     *
+     * $cloud = new ServiceBuilder([
+     *     'projectId' => 'my-awesome-project'
+     * ]);
+     *
+     * $vision = $cloud->vision();
+     *
+     * // The Vision client can also be instantianted directly.
+     * use Google\Cloud\Vision\VisionClient;
+     *
+     * $vision = new VisionClient([
+     *     'projectId' => 'my-awesome-project'
+     * ]);
+     * ```
+     *
+     * @param array $config {
+     *     Configuration Options.
+     *
+     *     @type string $projectId The project ID from the Google Developer's
+     *           Console.
+     *     @type callable $authHttpHandler A handler used to deliver Psr7
+     *           requests specifically for authentication.
+     *     @type callable $httpHandler A handler used to deliver Psr7 requests.
+     *     @type string $keyFile The contents of the service account
+     *           credentials .json file retrieved from the Google Developers
+     *           Console.
+     *     @type string $keyFilePath The full path to your service account
+     *           credentials .json file retrieved from the Google Developers
+     *           Console.
+     *     @type int $retries Number of retries for a failed request. Defaults
+     *           to 3.
+     *     @type array $scopes Scopes to be used for the request.
+     * }
+     * @throws \InvalidArgumentException
+     */
+    public function __construct(array $config = [])
+    {
+        if (!isset($config['scopes'])) {
+            $config['scopes'] = [self::FULL_CONTROL_SCOPE];
+        }
+
+        $this->connection = new Rest($this->configureAuthentication($config));
+    }
+
+    public function image($image, array $features = [], array $options = [])
+    {
+        return new Image($image, $features, $options);
+    }
+
+    public function images(array $images, array $features = [], array $options = [])
+    {
+        $result = [];
+        foreach ($images as $image) {
+            $result[] = $this->image($image, $features, $options);
+        }
+
+        return $result;
+    }
+
+    public function annotate(Image $image, array $options = [])
+    {
+        return $this->annotateBatch([$image], $options);
```

Comment:
I've thought about that some. I am fine with that approach, but if we go that way it should be implemented in all our APIs. Currently we're simply returning the response as it comes from the server everywhere. (see [`PubSub\Topic::publish()`](https://github.com/GoogleCloudPlatform/gcloud-php/blob/master/src/PubSub/Topic.php#L255-L288) for another example.)

I'd prefer to address it holistically if we decide that is the best approach!